components/staging/staging.js

```diff
@@ -330,11 +332,18 @@ FileViewModel.prototype.toggleStaged = function() {
 }
 FileViewModel.prototype.discardChanges = function() {
   var self = this;
-  var diag = components.create('yesnodialog', { title: 'Are you sure you want to discard these changes?', details: 'This operation cannot be undone.'});
-  diag.closed.add(function() {
-    if (diag.result()) self.server.post('/discardchanges', { path: self.staging.repoPath, file: self.name() });
-  });
-  programEvents.dispatch({ event: 'request-show-dialog', dialog: diag });
+  if (ungit.config.disableDiscardWarning || new Date().getTime() - this.staging.mutedTime < muteGraceTimeDuration) {
+    self.server.post('/discardchanges', { path: self.staging.repoPath, file: self.name() });
+  } else {
+    var diag = components.create('yesnomutedialog', { title: 'Are you sure you want to discard these changes?', details: 'This operation cannot be undone.'});
+    diag.closed.add(function() {
+      if (diag.result()) self.server.post('/discardchanges', { path: self.staging.repoPath, file: self.name() });
+      if (diag.result() === "mute") self.staging.mutedTime = new Date().getTime();
+
+      console.log(self.staging.mutedTime);
```

Comment:
I would remove the `console.log` or add a more descriptive message to it.
What do you think @codingtwinky ?